File: php/features.php
<?php
declare(strict_types=1);

// FreeUvIndex SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class FreeUvIndexFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new FreeUvIndexBaseFeature();
            case "test":
                return new FreeUvIndexTestFeature();
            default:
                return new FreeUvIndexBaseFeature();
        }
    }
}

File: py/features.py
# FreeUvIndex SDK feature factory

from feature.base_feature import FreeUvIndexBaseFeature
from feature.test_feature import FreeUvIndexTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FreeUvIndexBaseFeature(),
        "test": lambda: FreeUvIndexTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
